xtrack/base_element.py

```diff
@@ -16,25 +16,33 @@
 from .internal_record import RecordIdentifier, RecordIndex, generate_get_record
 
 start_per_part_block = """
-   int64_t const n_part = LocalParticle_get__num_active_particles(part0); //only_for_context cpu_serial cpu_openmp
-   #pragma omp parallel for                                       //only_for_context cpu_openmp
-   for (int jj=0; jj<n_part; jj+=!!CHUNK_SIZE!!){                 //only_for_context cpu_serial cpu_openmp
-    //#pragma omp simd
-    for (int iii=0; iii<!!CHUNK_SIZE!!; iii++){                   //only_for_context cpu_serial cpu_openmp
-      int const ii = iii+jj;                                      //only_for_context cpu_serial cpu_openmp
-      if (ii<n_part){                                             //only_for_context cpu_serial cpu_openmp
-
-        LocalParticle lpart = *part0;//only_for_context cpu_serial cpu_openmp
-        LocalParticle* part = &lpart;//only_for_context cpu_serial cpu_openmp
-        part->ipart = ii;            //only_for_context cpu_serial cpu_openmp
-
-        LocalParticle* part = part0;//only_for_context opencl cuda
+    {
+    const int64_t part_id = part0->ipart;  //only_for_context cpu_openmp
+    const int64_t end_id = part0->endpart; //only_for_context cpu_openmp
```

Comment:
These names are a bit confusing, as they have nothing to do with the `particle_id`